### PR TITLE
🛤️ Add API router for mailview UI

### LINKED ISSUE #8: 🛤️ API router
## Summary
Implement the internal API routes for the mailview UI.

## Endpoints
| Method | Path | Description |
|--------|------|-------------|
| GET | `/_mail/api/emails` | List all captured emails |
| GET | `/_mail/api/emails/{id}` | Get single email |
| GET | `/_mail/api/emails/{id}/html` | Get HTML body (for iframe) |
| GET | `/_mail/api/emails/{id}/attachments/{filename}` | Download attachment |
| DELETE | `/_mail/api/emails` | Clear all emails |
| DELETE | `/_mail/api/emails/{id}` | Delete single email |

## Acceptance Criteria
- [ ] `mailview/router.py` with Starlette routes
- [ ] JSON responses for list/get
- [ ] Proper content-types for HTML and attachments
- [ ] Unit tests for each endpoint

## Dependencies
- #6 SQLite store layer

### PR DESCRIPTION
**Pull request by swmcc**
## Summary
- Implements `MailviewRouter` class with Starlette routes for the mailview UI
- All endpoints for listing, viewing, and managing captured emails
- JSON responses for list/get, proper content-types for HTML and attachments

## Endpoints
| Method | Path | Description |
|--------|------|-------------|
| GET | `/emails` | List all captured emails (without bodies) |
| GET | `/emails/{id}` | Get single email with full content |
| GET | `/emails/{id}/html` | Get HTML body for iframe rendering |
| GET | `/emails/{id}/attachments/{filename}` | Download attachment |
| DELETE | `/emails` | Clear all emails |
| DELETE | `/emails/{id}` | Delete single email |

## Test plan
- [x] 19 unit tests covering all endpoints
- [x] Tests for success and error cases (404s)
- [x] Tests for edge cases (empty list, no HTML body)

Closes #8

### CODE REVIEW

File: mailview/router.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    async def list_emails(self, request: Request) -> JSONResponse:
+        """List all captured emails.
+
+        Returns JSON array of email summaries (without bodies).
+        """
+        emails = await self.store.get_all()
+        return JSONResponse(
+            {"emails": [email.to_dict(include_bodies=False) for email in emails]}
+        )
```
[the remark this answers is not in the document]
Comment (swmcc) [reply mailview/router.py]: Fixed - attachments field now removed from list response since get_all() doesn't populate it.

File: mailview/router.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+        if attachment is None:
+            return JSONResponse({"error": "Attachment not found"}, status_code=404)
+
+        return Response(
+            content=attachment.content,
+            media_type=attachment.content_type,
+            headers={
+                "Content-Disposition": f'attachment; filename="{attachment.filename}"'
+            },
```
[the remark this answers is not in the document]
Comment (swmcc) [reply mailview/router.py]: Fixed - filenames now sanitized by removing CR/LF and escaping quotes.
